### PR TITLE
git hooks

### PR DESCRIPTION
**Pull request by Shourya742**
Add pre-push git hooks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ To promote the adoption of Stratum V2, a comprehensive benchmarking tool is need
 ## 🐳 Prerequisites
 
 1. Install Docker on your system: https://docs.docker.com/engine/install/
+
+# Enable pre-push hooks
+This tells git where the githooks are located
+`# git config core.hooksPath .githooks`
```
[the remark this answers is not in the document]
Comment (Shourya742) [reply README.md:45]: how we gonna let other devs know how to configure them? if we dont mention in the readme ?

[the remark this answers is not in the document]
Comment (GitGab19) [reply README.md:45]: Sure, but I would like to see it written in a CONTRIBUTING.md file for example. Do you think it make sense?

[the remark this answers is not in the document]
Comment (Shourya742) [reply README.md:45]: Ya!!! Makes sense. Should I create a contributing.md file as a part of this PR?

[the remark this answers is not in the document]
Comment (GitGab19) [reply README.md:45]: Yeah I think it's the best thing to do 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<!-- omit in toc -->
+# Contributing to the Benchmarking Tool
+
+First off, thank you for taking the time to contribute! ❤️
+
+All types of contributions are encouraged and valued. Please refer to the [Table of Contents](#table-of-contents) for various ways you can help and details on how this project handles contributions. Be sure to read the relevant sections before contributing—it will make things easier for us maintainers and provide a smoother experience for everyone involved. The community looks forward to your contributions! 🎉
```
Comment (GitGab19) [inline CONTRIBUTING.md:6]: typo in "contributing-it"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  - [Contribution Workflow](#contribution-workflow)
+  - [Your First Code Contribution](#your-first-code-contribution)
+  
+
+## I Have a Question
+
+> Before asking a question, please ensure you’ve read the documentation available in the `docs` directory.
+
+The best way to ask a question is to join our community [Discord](https://discord.com/invite/fsEW23wFYs). The most suitable place to post your question is:
+- #benchmarking-tool
```
Comment (GitGab19) [inline CONTRIBUTING.md:29]:
```suggestion
The best way to ask a question is by joining our community on [Discord](https://discord.com/invite/fsEW23wFYs). The most suitable place to post your question is `#benchmarking-tool`
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+7. **Merge and Close**: Once your pull request has been approved and all discussions have been resolved, a project maintainer will merge your changes into the `main` branch. Your contribution will then officially become part of the project. The pull request will be closed, marking the completion of your contribution.
+
+### Your First Code Contribution
+
+> To contribute, a basic understanding of Git and GitHub is needed. If you're not familiar with them, check out [this resource](https://docs.github.com/en/get-started/start-your-journey/git-and-github-learning-resources) to learn how to use them.
+
+Not sure where to start contributing to SRI? You can begin by looking through issues labeled `good first issue`:
+
+* [Good first issue](https://github.com/stratum-mining/benchmarking-tool/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) - These issues typically require only a few lines of code and a test or two.
```
Comment (GitGab19) [inline CONTRIBUTING.md:76]: I would add also the `help-wanted` label.

And I'd also reference the project roadmap.

I'm just comparing this with the `contributing.md` we have on `stratum` repo: https://github.com/stratum-mining/stratum/blob/0f0ee1e32a7cdeb9af83718364e0896b36744824/CONTRIBUTING.md?plain=1#L109

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+Our development calls are scheduled every Tuesday at 18:00 CET. You can see them in the Discord sidebar under Events and subscribe to be notified.
+
+## I Want To Contribute
+
+> When contributing to this project, you must agree that you have authored 100% of the content, that you have the necessary rights to the content, and that the content you contribute may be provided under the project license.
+
+### Contribution Workflow
+
+The SRI project follows an open contributor model, where anyone is welcome to contribute through reviews, documentation, testing, and patches. Follow these steps to contribute:
```
Comment (GitGab19) [inline CONTRIBUTING.md:52]: I wouldn't use SRI here. I'd use just "The project" follows...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+2. **Set Up Git Hooks**
+
+    `git config core.hooksPath .githooks`: This command configures Git to use the `.githooks` directory for all Git hooks.
+
+3. **Create a Branch**
+
+4. **Commit Your Changes**
+
+5. **Submit a Pull Request**: Once you're satisfied with your changes, submit a pull request to the original SRI repository. Provide a clear and concise description of the changes you've made. If your pull request addresses an existing issue, reference the issue number in the description. To contribute to the protocol implementation, every PR must be opened against the `dev` branch. To better understand the SRI branch structure, please refer to [this guide](https://github.com/stratum-mining/stratum/blob/main/RELEASE.md#principal-branches).
```
Comment (GitGab19) [inline CONTRIBUTING.md:64]: Same here (with regards to SRI)

Comment (GitGab19) [inline CONTRIBUTING.md:64]: Typo: `dev` should be `main`

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -74,3 +73,7 @@ The SRI project follows an open contributor model, where anyone is welcome to co
 Not sure where to start contributing to SRI? You can begin by looking through issues labeled `good first issue`:
 
 * [Good first issue](https://github.com/stratum-mining/benchmarking-tool/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) - These issues typically require only a few lines of code and a test or two.
+
+* [Help wanted](https://github.com/stratum-mining/benchmarking-tool/labels/help%20wanted) - issues which should be a bit more involved than good first issue issues.
+
+Another way to better understand where to focus your contribution is by looking at our roadmap: https://docs.google.com/document/d/1CqcvsxGugFjWy4e4Yf6PjxCs2O4puwlFBO6M0TRL4qE/edit#heading=h.adukd8mh74q7
```
Comment (GitGab19) [inline CONTRIBUTING.md:79]:
```suggestion
Another way to better understand where to focus your contribution is by looking at our project board: https://github.com/orgs/stratum-mining/projects/10
```

**Review by GitGab19 — APPROVED** [review]

**Pull request merged**